nixos/modules/virtualisation/containers.nix

```diff
@@ -318,6 +319,16 @@ let
       '';
     };
 
+    hostPort = mkOption {
+      type = types.nullOr types.string;
```

Comment:
First, now I found that `types.str` is the correct way, thanks for pointing this out.
The description is misleading. I will change it. 
I have to check whether `systemd-nspawn` can take multiple `--port` parameters. If so, it's much desirable to be a list. It would be better to be a list of key/values where key = protocol,hostPort,containerPort. Thanks for suggestion.